platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SavedStateMetadata.java

```diff
@@ -241,18 +278,46 @@ private static void logMissingField(final SavedStateMetadataField field) {
      * @param value the invalid value
      * @param e     the exception
      */
-    private static void logInvalidField(final SavedStateMetadataField field, final String value, final Exception e) {
+    private static void logInvalidField(
+            @NonNull final SavedStateMetadataField field, @NonNull final String value, @NonNull final Exception e) {
         logger.warn(
                 STARTUP.getMarker(), "Signed state metadata file has invalid value for field {}: {}", field, value, e);
     }
 
+    /**
+     * Throw an exception for a missing required field.
+     *
+     * @param field the missing field
+     */
+    private static void throwMissingRequiredField(@NonNull final SavedStateMetadataField field) throws IOException {
+        throw new IOException("Signed state metadata file is missing required field: " + field);
```

Comment:
I get we're already throwing an exception, but if the argument to the method call truly requires `@NonNull`, this is not going to tell us that there was an incorrect value passed in.  Should an NPE supersede?

Reply:
The reason why it needs to throw an `IOException` is because we catch and handle exceptions encountered while parsing the metadata file:

```
try {
    metadata = SavedStateMetadata.parse(metdataPath);
} catch (final IOException e) {
    logger.error(
            EXCEPTION.getMarker(), "Unable to read saved state metadata file '{}'", metdataPath);
    continue;
}
```

This is an interesting edge case that I had not previously considered before. Does the annotation `@NonNull` mean "pass in null and get a nullptr exception", or does it mean "don't pass in null else bad things might happen"?

I will raise this issue with the larger group to discuss.

Reply:
added `Objects.requireNonNull(field);`